src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/impl/NoActionProcessor.java

```diff
@@ -1,30 +1,30 @@
-package uk.gov.ons.census.fwmt.outcomeservice.converter.spg;
+package uk.gov.ons.census.fwmt.outcomeservice.converter.impl;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.ons.census.fwmt.common.error.GatewayException;
 import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
-import uk.gov.ons.census.fwmt.outcomeservice.converter.SpgOutcomeServiceProcessor;
-import uk.gov.ons.census.fwmt.outcomeservice.dto.SpgOutcomeSuperSetDto;
+import uk.gov.ons.census.fwmt.outcomeservice.converter.OutcomeServiceProcessor;
+import uk.gov.ons.census.fwmt.outcomeservice.dto.OutcomeSuperSetDto;
 
 import java.util.UUID;
 
 import static uk.gov.ons.census.fwmt.outcomeservice.config.GatewayEventsConfig.RECEIVED_NO_ACTION_FROM_TM;
 
 @Component("NO_ACTION")
-public class SpgNoActionProcessor implements SpgOutcomeServiceProcessor {
+public class NoActionProcessor implements OutcomeServiceProcessor {
 
   @Autowired
   private GatewayEventManager gatewayEventManager;
 
   @Override
-  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+  public UUID process(OutcomeSuperSetDto outcome, UUID caseIdHolder, String type) throws GatewayException {
     gatewayEventManager
         .triggerErrorEvent(this.getClass(), (Exception) null, "Action not expected",
-            RECEIVED_NO_ACTION_FROM_TM,
+            RECEIVED_NO_ACTION_FROM_TM, "NO ACTION",
```

Comment:
This might cause an issue as the meta data is key/value
and you might have 
NoAction/Transaction-id
String.valueOf(outcome.getTransactionId())/Primary Outcome
etc